### ISSUE TITLE
Selector switch issue with more then 31 levels

### ISSUE THREAD

**Issue by kregory** [domoticz/domoticz#5324]
My switch selector handle 31 differents states.
Passing the 25th state, levels are not the ones that are parametrized 

level 250 show 4 as level
level 260 show 14 as level
level 270 show 24 as level
...

**Comment by gizmocuz** [comment]
Thank you for spotting this.
If I open the browsers developers console and log the network traffic, I can see that when I select the 26 item, two json posts are fired, one with the correct one, and one with level 31, level 31 is also selected after this

**Comment by gizmocuz** [comment]
I think this is caused by the level action values, they should not go higher then 255 as internally we use a uint8_t as level

**Comment by odelma** [comment]
Yeah, something I've noted as well and needed to create a special handler for these levels in my lua scripts. As such, they have been working reliable with the handler. 

Please let us know if/when this changes so that I know to take away the handler from the scripts 😄

**Comment by kregory** [comment]
How do you deal with it ?

**Comment by gizmocuz** [comment]
Maybe do not use a selector switch with more than 31 values.... that is crazy anyway, what device do you control with this ?

**Comment by kregory** [comment]
> Maybe do not use a selector switch with more than 31 values.... that is crazy anyway, what device do you control with this ?

Sorry my answer was for odelma

Anyway it's to control percentage difference between insuflation and extraction motor on a controled ventilation. Values accepted are from -15% to +15%

**Comment by odelma** [comment]
I use it to select the tv channel 😅

**Comment by odelma** [comment]
> How do you deal with it ?

I can check the script later and share it with you 👍

**Comment by gizmocuz** [comment]
@odelma , I know there is somewhere a remote control in Domoticz as well
But actually, controlling channels/volume is not meant to be done by Domoticz, turning it on/off/mute/changing actions like tv/netflix/etc  I understand

**Comment by odelma** [comment]
> @odelma , I know there is somewhere a remote control in Domoticz as well But actually, controlling channels/volume is not meant to be done by Domoticz, turning it on/off/mute/changing actions like tv/netflix/etc I understand

That is certainly true (that it is not an intended use case), but nonetheless it is a very convenient way of setting the TV channel 👍 

Having to use the lua handler for this is not a problem at all so everything's good. But if the logic will change I am hoping there's a heads up for us (well, I'll find out soon enough in any case when handler stops working 😄). So from my viewpoint this was not a request to get 10+ selector levels supported, I just wanted to acknowledge to the original poster that I have noticed this as well and there are workarounds available. In some cases I assume 10+ selector levels can be justified so in theory this could cause issues with some plugins for example, but have not noticed anything like this.

**Comment by odelma** [comment]
> How do you deal with it ?

I use this in lua script:

```
local Selection = otherdevices_svalues['Kanavat']
local Special = {"4", "14", "24", "34", "44", "54", "64", "74"} --, "84", "94", "104", "114"} -> add more if needed

for index, value in pairs(Special) do
    if value == Selection then
        isSpecial = true
    end
end

if isSpecial then
    if Selection == "4" then
        Channel = "26"
    else
        Channel = tostring(tonumber(Selection:sub(1, 1)) + 26)
    end
else
    Channel = Selection:sub(1, -2)
end

if Channel ~= "33" and (otherdevices['Median tila'] == "TV" or otherdevices['Median tila'] == "STV") then
    if otherdevices['Median tila']  == 'TV' then
        if tonumber(Channel) < 10 then
             os.execute('sudo /home/pi/HarmonyHubControl/HarmonyHubControl 192.168.11.157 issue_device_command 28537515 '..Channel..' OK &')
        else
            local firstNum = Channel:sub(1, 1)
            local secondNum = Channel:sub(-1)
             os.execute('sudo /home/pi/HarmonyHubControl/HarmonyHubControl 192.168.11.157 issue_device_command 28537515 '..firstNum..' '..secondNum..' OK &')
        end
    elseif otherdevices['Median tila']  == 'STV' then
        if tonumber(Channel) < 10 then
            os.execute('sudo /home/pi/HarmonyHubControl/HarmonyHubControl 192.168.11.157 issue_device_command 55813518 '..Channel..' OK &')
        else
            local firstNum = Channel:sub(1, 1)
            local secondNum = Channel:sub(-1)
            os.execute('sudo /home/pi/HarmonyHubControl/HarmonyHubControl 192.168.11.157 issue_device_command 55813518 '..firstNum..' '..secondNum..' OK &')
        end
    end
    commandArray[#commandArray+1]={['UpdateDevice'] = ''..otherdevices_idx['Kanavat']..'|0|330'}
end
```

So this is basically just checking what is the svalue from the selector and when we go over certain selector level (24th -> 240, 25th -> 4) the svalues change to this unexpected series 4, 14, 24, 34... So I am checking if svalue is one of these "special" values and if yes, then there is a logic to determine which channel that represents (note, to pass a tv channel I need to convert the "normal" selector levels 10, 20, 30... to channel values 1, 2, 3...). Far from an elegant solution, I am sure, but it works so good enough for personal use... 😅

**Comment by gizmocuz** [comment]
A switch with 31 options is not the way to go and it is better to rethink the design
